### PR TITLE
Add new rule: `no-class-bindings`

### PR DESCRIPTION
**Pull request by scalvert**
Implements #1762

### CODE REVIEW

File: docs/rule/no-class-bindings.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# no-class-bindings
+
+It is possible to pass `classBinding` and `classNameBindings` as arguments to a component when invoked with curly syntax.
+
+```hbs
+{{some-component classNameBindings="foo:truthy:falsy"}}
+```
+
+These arguments are merged into the class attribute on the class, regardless of whether or not the component is a classic component which contains the classNameBindings logic.
```
Comment (scalvert) [inline docs/rule/no-class-bindings.md:9]: I'm open to a better description. This one sounds janky.

File: docs/rule/no-class-bindings.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+This rule disallows components accepting either `classBinding` or `classNameBindings`.
```
Comment (MelSumner) [inline docs/rule/no-class-bindings.md:11]:
```suggestion
This rule disallows use of `classBinding` or `classNameBindings` on components.
```

File: docs/rule/no-class-bindings.md (hunk 1 above)
Comment (MelSumner) [inline docs/rule/no-class-bindings.md:9]:
```suggestion
These arguments are merged into the `class` attribute, whether or not the component is a classic component (which contains the `classNameBindings` logic).
```

**Review by MelSumner — APPROVED** [review]
A few prose suggestions but not blocking. 
Do you think we should include a `--fix` option?

Thank you for working on this issue! 👍

**Comment by rwjblue** [comment]
> Do you think we should include a --fix option?

I don't know how to fix all cases (specifically `classNameBindings`), but we could fix `classNames` probably (refactoring in curlies to `class=` is all you do, in elements you'd `@classNames` -> `class=` as an attribute).

**Comment by scalvert** [comment]
@rwjblue it's `classBinding` and `classNameBindings`, there is no `classNames` in this PR.

**Comment by scalvert** [comment]
@MelSumner we could, but I'm unsure we can cover all cases. I'd prefer to avoid unless we're sure we can fix reasonably.

**Review by rwjblue — APPROVED** [review]

**Pull request merged**